### PR TITLE
global: remove ETag/Last-Modified headers

### LINKED ISSUE #851: Physical Item: status not consistent with loan
## Steps to Reproduce

1. Go to Loan details page (Pending)
2. Checkout one copy.
3. Click on Physical copy link
4. The status displayed is not consistent with the loan (item still visible as on shelf)

## The expected behavior

The status should be consistent with the loan. 

Possibly the error is caused by the indexer not working properly

### LINKED ISSUE #852: Document details page - statistics on top incorrect
## Steps to Reproduce

1. Go to a pending loan
2. Checkout any available item
3. Go to the document details page (of this loan)
4. See the available items number on top (statistics)

## The current behavior

The statistics are not updated correctly (f.e. number of active loans is not correct)

## The expected behavior

The statistics of the document should be up to date with the actual catalogue situation

Possibly indexer issue 
Might be related to https://github.com/inveniosoftware/invenio-app-ils/issues/851

### LINKED ISSUE #857: Loan checkout: physical copy not updated
## Description

When performing checkout from f.e. patron details page, the loan status is not updated.

## Steps to Reproduce

1. Go to Patron details page
2. Go to pending loan requests, click on one 
3. Check out the loan
4. see the empty "physical item" field, see the loan status
4. go to the Item details page (the one you've checked out) the status of the item is not updated

## The current behavior

Physical Item was not updated. Displayed status of the item s still "on shelf". Loan status was not updated.

## The expected behavior

The physical copy field should be updated, the loan status should be updated, "on shelf" on the physical copy details page should be changed to "on loan" 

Probably indexing issue, related to: 
https://github.com/inveniosoftware/invenio-app-ils/issues/851

### LINKED ISSUE #861: Document not reindexed after changing the state of physical item
Software version (if known):

## Describe the bug

The status is correctly changed, and is ok on the BackOffice, but the item is still visible as "On shelf" on the Frontsite.

## Steps to Reproduce

1. Change status from available to missing of any physical item
2. Go to the frontsite view of the document which contains this item
3. See the status of the physical item listed for the document

## The current behavior

the status is not updated from "on shelf" to missing (or removed from the list - check the initial search query)

## The expected behavior

status should be updated - probably the indexing error (document not reindexed after item changed)

### PR DESCRIPTION
**Pull request by ntarocco**
* removes the ETag/Last-Modifies headers when returning the records
  REST responses to prevent caching of extra fields added at
  serialization time.
* closes #851
* closes #852
* closes #857
* closes #861

### CODE REVIEW

**Comment by ntarocco** [comment]
Not very happy with the solution, in particular in the case of `PUT` requests. But I couldn't find anything better.
I can explain more about the `PUT` requests issue by voice.

File: invenio_app_ils/records/serializers/__init__.py [hunk 1]
```diff
@@ -7,27 +7,54 @@
...
+    However, it looks like that browsers are not sending the If-Match/ETag
+    headers on PUT requests.
+    """
+
+    def view(pid, record, code=200, headers=None, links_factory=None):
+        response = current_app.response_class(
+            serializer.serialize(pid, record, links_factory=links_factory),
+            mimetype=mimetype)
+        response.status_code = code
+        response.cache_control.no_cache = True
```
Comment (kpsherva) [inline invenio_app_ils/records/serializers/__init__.py:39]: is it possible to have a test for this?

**Review by kpsherva — APPROVED** [review]

File: invenio_app_ils/records/serializers/__init__.py (hunk 1 above)
Comment (ntarocco) [reply invenio_app_ils/records/serializers/__init__.py:39]: very good point, I have added it.

**Review by topless — APPROVED** [review]
Great, I would be interested to hear more about the `PUT` requests.

**Pull request merged**